File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.py
class Solution(object):
    def maximumUnits(self, boxTypes, truckSize):
        """
        :type boxTypes: List[List[int]]
        :type truckSize: int
        :rtype: int
        """
        boxTypes.sort(key = lambda x: -x[1])
        maxUnit = 0
        for num, unit in boxTypes:
            maxUnit += unit * min(num, truckSize)
            truckSize -= min(num, truckSize)
        return maxUnit
            

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a, b) -> (b[1] - a[1]));
        int maxUnit = 0;
        for (int[] boxType : boxTypes) {
            int num = boxType[0], unit = boxType[1];
            maxUnit += unit * Math.min(num, truckSize);
            truckSize -= Math.min(num, truckSize);
            if (truckSize == 0) {break;}
        }
        return maxUnit;
    }
}